src/createXcmCalls/limitedTeleportAssets.ts

```diff
@@ -31,26 +31,32 @@ export const limitedTeleportAssets = (
 	xcmVersion: number,
 	specName: string,
 	registry: Registry,
+	transferForeignAssets: boolean | undefined,
 	weightLimit?: string,
 	paysWithFeeDest?: string
-): SubmittableExtrinsic<'promise', ISubmittableResult> => {
+): Promise<SubmittableExtrinsic<'promise', ISubmittableResult>> => {
 	const pallet = establishXcmPallet(api);
 	const ext = api.tx[pallet].limitedTeleportAssets;
 	const typeCreator = createXcmTypes[direction];
 	const beneficiary = typeCreator.createBeneficiary(api, destAddr, xcmVersion);
 	const dest = typeCreator.createDest(api, destChainId, xcmVersion);
-	const assets = typeCreator.createAssets(
+	const assets = await typeCreator.createAssets(
 		api,
 		normalizeArrToStr(amounts),
 		xcmVersion,
 		specName,
 		assetIds,
-		{ registry }
+		{ registry },
+		transferForeignAssets
```

Comment:
1. Can we name `transferForeignAssets` to `isForeignAssetsTransfer`. This would allow things to make sense with the boolean. 

2. Then `isForeignAssetsTransfer` should be under the options field along with the registry. This is because when we add liquid tokens, there will be a field as well that says `isLiquidToken` transfer, and it will allow us to organize our boolean logic for which kind of transfer it should be.

Reply:
perhaps I am getting this wrong, but with this design, are we adding a new optional input to specify if the user wants to transfer a foreign asset? what is the motivation behind this decision? A user transferring a foreign asset should have the same UX as transferring native assets. The API can know it is a foreign asset as soon as the asset ID input is a multilocation and not a simple integer. Is it possible to avoid the extra optional value and have the same UX as for native assets? 

Note that this is different than with pool assets where the same ID system as normal assets is used and hence it may get confusing. Plus it is a way less critical functionality so it is ok to add the extra UX step.

Reply:
yes, the UX will be the same for Assets and ForeignAssets for the user